File: app/src/main/java/lee/jaebeom/allwiki/main/MainTabPresenter.java
package lee.jaebeom.allwiki.main;

import java.io.Serializable;

import lee.jaebeom.allwiki.wiki.WikiData;

/**
 * Created by leejaebeom on 2017. 10. 15..
 */

public class MainTabPresenter implements MainContract.TabPresenter, Serializable {
    String HTML;
    private final MainContract.TabView tabView;

    public MainTabPresenter(MainContract.TabView tabView) {
        this.tabView = tabView;
        this.tabView.setPresenter(this);
    }

    @Override
    public void CallBack() {
        tabView.CallBack();
    }

    @Override
    public void start() {

    }

    @Override
    public void saveHTML(String Html) {
        HTML = Html;
    }

    @Override
    public String getHTML() {
        return HTML;
    }

    @Override
    public String getWikiURL(String wiki){
        return WikiData.wikiURL.get(wiki).getURL();
    }

    @Override
    public String getWikiFrontURL(String wiki) {
        return WikiData.wikiURL.get(wiki).getFrontURL();
    }
}

File: app/src/main/java/lee/jaebeom/WikiPortal/main/MainTabPresenter.kt
package lee.jaebeom.WikiPortal.main

import lee.jaebeom.WikiPortal.wiki.WikiData
import java.io.Serializable

/**
 * Created by leejaebeom on 2017. 11. 6..
 */
class MainTabPresenter(tabView: MainContract.TabView) : MainContract.TabPresenter, Serializable{
    lateinit var HTMLCode : String
    private var tabView: MainContract.TabView = tabView

    init {
        this.tabView._presenter(this)
    }
    override fun start() {

    }

    override fun saveHTML(Html: String) {
        HTMLCode = Html
    }

    override fun getHTML(): String {
        return HTMLCode
    }

    override fun CallBack() {
        tabView.CallBack()
    }

    override fun getWikiURL(wiki: String): String {
        return WikiData.wikiURL.get(wiki)!!.URL
    }

    override fun getWikiFrontURL(wiki: String): String {
        return WikiData.wikiURL.get(wiki)!!.frontURL
    }

}